rmgpy/solver/base.pyx

```diff
@@ -224,18 +224,57 @@ cdef class ReactionSystem(DASx):
         self.sensitivityCoefficients = numpy.zeros((self.numCoreSpecies, self.numCoreReactions), numpy.float64)
         self.unimolecularThreshold = numpy.zeros((self.numCoreSpecies), bool)
         self.bimolecularThreshold = numpy.zeros((self.numCoreSpecies, self.numCoreSpecies), bool)
+
+        self.initialize_surface(coreSpecies,coreReactions,surfaceSpecies,surfaceReactions)
         
-        self.surfaceSpeciesIndices = numpy.zeros(len(surfaceSpecies),dtype=numpy.int)
-        self.surfaceReactionIndices = numpy.zeros(len(surfaceReactions),dtype=numpy.int)
         
-        for i in range(len(surfaceSpecies)):
-            self.surfaceSpeciesIndices[i] = coreSpecies.index(surfaceSpecies[i])
-        for i in range(len(surfaceReactions)):
-            self.surfaceReactionIndices[i] = coreReactions.index(surfaceReactions[i]) 
-
     def initialize_solver(self):
         DASx.initialize(self, self.t0, self.y0, self.dydt0, self.senpar, self.atol_array, self.rtol_array)
-
+    
+    def initialize_surface(self,coreSpecies,coreReactions,surfaceSpecies,surfaceReactions):
+#        cdef list productIndices,reactantIndices, surfaceSpeciesIndices, surfaceReationIndices
+#        cdef set possibleSpeciesIndices
+#        cdef int i,j
+#        cdef bool notInSurface
+        
+        logging.info('initializing surface')
+        productIndices = self.productIndices
+        reactantIndices = self.reactantIndices
+        
+        surfaceSpeciesIndices = []
+        surfaceReactionIndices = []
+        possibleSpeciesIndices = set()
+        
+        for i in range(len(surfaceSpecies)):
+            surfaceSpeciesIndices.append(coreSpecies.index(surfaceSpecies[i]))
+        for i in range(len(surfaceReactions)):
+            surfaceReactionIndices.append(coreReactions.index(surfaceReactions[i]))
```

Comment:
Unless you're going to be doing something fancy like manipulating (popping or inserting) the list while iterating over it, it's more common (I think preferable) in Python to do loops like:
```python
for species in surfaceSpecies:
    do_something_with(species)
```
than
```python
for i in range(len(surfaceSpecies)):
    do_something_with(surfaceSpecies[i])
```
This pattern occurs quite a lot.

(unless this is part of some cython optimization I haven't thought of?)